go/nbs/table_set.go

```diff
@@ -96,6 +96,19 @@ func (ts tableSet) Prepend(mt *memTable) tableSet {
 	return tableSet{newTables, ts.p, ts.rl}
 }
 
+func (ts tableSet) extract(order EnumerationOrder, chunks chan<- extractRecord) {
+	// Since new tables are _prepended_ to a tableSet, extracting chunks in ReverseOrder requires iterating ts.chunkSources from front to back, while doing insertOrder requires iterating back to front.
+	if order == ReverseOrder {
+		for _, cs := range ts.chunkSources {
+			cs.extract(order, chunks)
+		}
+		return
+	}
+	for i := len(ts.chunkSources) - 1; i >= 0; i-- {
```

Comment:
so would I!

Go, however, apparently doesn't. Increment and decrement aren't expressions in Go (statements? whatever the right term is. You can't use them inline like that :-/)